### ISSUE TITLE
Support for null values

### ISSUE THREAD

**Issue by yonojoy** [landgraf-dev/openapi-delphi-generator#28]
We have a server that delivers the following JSON (for `TClient`):

	{
		"id": 6718,
		"first_name": "Hans",
		"last_name": "Bauer",
		"telephone": null,
		"email": null,
		"gender": null
	}

If I read this into a `TClient` variable and then send this to the server again, this is translated to

	{
		"id": 6718,
		"first_name": "Hans",
		"last_name": "Bauer",
		"telephone": "",
		"email": "",
		"gender": 0
	}

which the server does not like: "0 is not a valid value for gender".

There are two solutions to this problem:
1. Implement handling for `null` values
2. Provide an option to treat `null` as `not ...HasValue`

While I think 1 is the correct solution, 2 can be implemented more easily by modifying `OpenApiJson.pas`, for example for `USEDBX`:

	function TJsonWrapper.ObjContains(JObj: TJSONValue; const Name: string; out Value: TJSONValue): Boolean;
	var
	  Pair: TJSONPair;
	begin
	  Pair := TJSONObject(JObj).Get(Name);
	  if Assigned(Pair) then
	  begin
		Value := Pair.JsonValue;
		if Value.Null and FOptionTreatNullAsNotHasValue then
		  Value := nil;
	  end
	  else
		Value := nil;
	end;

Solution 2 would result in the following JSON, which the server happily processes:

	{
		"id": 6718,
		"first_name": "Hans",
		"last_name": "Bauer"
	}

@wlandgraf What do you think?

**Comment by wlandgraf** [comment]
@yonojoy I think the correct way is 1, to handle `null` values.
`HasValue` properties have a different meaning, and the meaning is exactly "is it present in JSON" (when reading) or "was it set from code" (when writing, which ends up in the property not being included in JSON).

So `null` should have a different treatment. I believe all nullable properties (and this should be rightfully inferred from the JSON schema) should have an additional `XxxIsNull` helper which indicates if the value is `null` or not.

And the property setter should set `FIsNull` to `False` the same way it sets `FHasValue` to `True`. The JSON serialization/deserialization will also get more complex.

**Comment by yonojoy** [comment]
@wlandgraf Yes, I completely agree with your analysis. But it's a big amount of work. I won't have the resources to implement this myself.

But for me and maybe for others, solution 2 solves the immediate problems. Therefore I would suggest to add a define `TREAT_NULL_VALUES_AS_NOT_HAS_VALUE`, that implements 2 until solution 1 will be available.
